// @devvit/protos/types/devvit/gateway/utils/v1alpha/app_health.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { CheckHealthRequest, CheckHealthResponse } from './app_health.js';
export var AppHealthMethod;
(function (AppHealthMethod) {
    AppHealthMethod["CheckHealth"] = "CheckHealth";
})(AppHealthMethod || (AppHealthMethod = {}));
export const AppHealthMethodList = [AppHealthMethod.CheckHealth];
export function createAppHealthServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.gateway.utils.v1alpha',
        serviceName: 'AppHealth',
        methodList: AppHealthMethodList,
        matchRoute: matchAppHealthRoute,
    });
}
function matchAppHealthRoute(method, events) {
    switch (method) {
        case 'CheckHealth':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'CheckHealth' };
                await events.onMatch(ctx);
                return handleAppHealthCheckHealthRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleAppHealthCheckHealthRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleAppHealthCheckHealthJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleAppHealthCheckHealthProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleAppHealthCheckHealthJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = CheckHealthRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.CheckHealth(ctx, inputReq);
        });
    }
    else {
        response = await service.CheckHealth(ctx, request);
    }
    return JSON.stringify(CheckHealthResponse.toJSON(response));
}
async function handleAppHealthCheckHealthProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = CheckHealthRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.CheckHealth(ctx, inputReq);
        });
    }
    else {
        response = await service.CheckHealth(ctx, request);
    }
    return Buffer.from(CheckHealthResponse.encode(response).finish());
}
